Doc/whatsnew/3.11.rst

```diff
@@ -70,6 +70,49 @@ Summary -- Release highlights
 New Features
 ============
 
+.. _whatsnew310-pep657:
+
+Enhanced error locations in tracebacks
+--------------------------------------
+
+When printing tracebacks, the interpreter will now point to the exact expression
+that caused the error instead of just the line. For example:
+
+.. code-block:: pycon
+
+    Traceback (most recent call last):
+      File "distance.py", line 11, in <module>
+        print(manhattan_distance(p1, p2))
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^
+      File "distance.py", line 6, in manhattan_distance
+        return abs(point_1.x - point_2.x) + abs(point_1.y - point_2.y)
+                               ^^^^^^^^^
+    AttributeError: 'NoneType' object has no attribute 'x'
+
+Previous versions of the interpreter would point to just the line making it
+ambiguous which object was ``None``. These enhanced errors can also be helpful
+when dealing with deeply nested dictionary objects:
+
+.. code-block:: pycon
+
+    Traceback (most recent call last):
+      File "json_handler.py", line 21, in <module>
+        json_blob['a']['b']['c']['d']['e']
+        ^^^^^^^^^^^^^^^^^^^^^^^^
+    TypeError: 'NoneType' object is not subscriptable
+
+This new column information for expressions can be retrieved using the
```

Comment:
Not sure if the `co_positions()` method and memory usage should be pointed out here.

Reply:
I don't think we should mention about that in the what's new, maybe just as a little warning in the `co_positions()`

Reply:
Moved to a note in `co_positions()`.